integration/datasource_test.go

```diff
@@ -12,6 +12,74 @@ import (
 )
 
 func TestDatasources(t *testing.T) {
+	dir := "testdata/datasources"
+	setupContexts(t, dir)
+
+	t.Run("Get datasource - success", func(t *testing.T) {
+		runTest(t, GrizzlyTest{
+			TestDir:       dir,
+			RunOnContexts: allContexts,
+			Commands: []Command{
+				{
+					Command:            "get Datasource.392IktgGk",
+					ExpectedCode:       0,
+					ExpectedOutputFile: "392IktgGk-returned.yaml",
+				},
+			},
+		})
+	})
+
+	t.Run("Get datasource list - success", func(t *testing.T) {
+		runTest(t, GrizzlyTest{
+			TestDir:       dir,
+			RunOnContexts: allContexts,
+			Commands: []Command{
+				{
+					Command:            "list -r -t Datasource",
+					ExpectedOutputFile: "list.txt",
+				},
+			},
+		})
+	})
+
+	t.Run("Apply datasource", func(t *testing.T) {
+		runTest(t, GrizzlyTest{
+			TestDir:       dir,
+			RunOnContexts: allContexts,
+			Commands: []Command{
+				{
+					Command:        "apply appdynamics2.yaml",
+					ExpectedOutput: "Datasource.appdynamics2 added\n",
+				},
+				{
+					Command:            "get Datasource.appdynamics2",
+					ExpectedOutputFile: "appdynamics2-returned.yaml",
+				},
+			},
+		})
+	})
+
+	t.Run("Apply broken datasource", func(t *testing.T) {
+		runTest(t, GrizzlyTest{
+			TestDir:       dir,
+			RunOnContexts: allContexts,
+			Commands: []Command{
+				{
+					Command:        "apply datasource-broken.json",
+					ExpectedOutput: "Datasource.datasource-broken added\n",
+				},
+				{
+					Command: "get -o json Datasource.datasource-broken",
+					ExpectedOutputFiles: []string{
+						"datasource-broken-returned.json",
+						"datasource-broken-returned2.json",
```

Comment:
I don't understand. Why is there the same file twice with a different name?

Reply:
because the subdir instance returns different data. But for now I've just excluded the subdir target for this test for this reason.